File: spec/unit/GameSpec.js
'use strict';

describe('Game', function() {

  let game;

  beforeEach(function() {
    game = new Game();
  });

  describe('initialisation of a new Game', function() {
    it('creates an empty frameScores array', function() {
      expect(game.frameScores).toEqual([]);
    });
  });

  describe('Recording frame points after frame is terminated', function() {
    it('appends the frame\'s points to the frameScores array for no strike or spare', function() {
      game.play(7, 2);
      expect(game.frameScores).toContain(9);
    });

    it('appends the frame\'s points to the frameScores array in the case of a strike', function() {
      game.play(10);
      expect(game.frameScores).toContain(10);
      expect(game.strike).toBe(true);
    });

    it('appends the frame\'s points to the frameScores array in the case of a strike', function() {
      game.play(8, 2);
      expect(game.frameScores).toContain(10);
      expect(game.spare).toBe(true);
    });
  });
});

File: spec/unit/game_spec.rb
require 'game'

describe Game do

  before do
    @game = Game.new
  end

  describe '#first_roll' do   
    it 'stores the pins hit in an array' do
      @game.first_roll(4)
      expect(@game.pins_down).to eq [4]
    end

    it 'stores an X for a strike' do
      @game.first_roll(10)
      expect(@game.pins_down).to eq ["X"]
    end
  end

  describe '#second_roll' do   
    it 'stores the pins hit in an array' do
      @game.first_roll(4)
      @game.second_roll(5)
      expect(@game.pins_down).to eq [4,5]
    end

    it 'stores an / for a spare' do
      @game.first_roll(4)
      @game.second_roll(6)
      expect(@game.pins_down).to eq [4,"/"]
    end
  end


    

end
